contracts/pods/PodAdminGateway.sol

```diff
@@ -0,0 +1,219 @@
+import {TimelockController} from "@openzeppelin/contracts/governance/TimelockController.sol";
+import {IMemberToken} from "./orcaInterfaces/IMemberToken.sol";
+import {CoreRef} from "../refs/CoreRef.sol";
+import {ICore} from "../core/ICore.sol";
+import {IControllerV1} from "./orcaInterfaces/IControllerV1.sol";
+import {Core} from "../core/Core.sol";
+import {TribeRoles} from "../core/TribeRoles.sol";
+import {IPodAdminGateway} from "./IPodAdminGateway.sol";
+import {IPodFactory} from "./IPodFactory.sol";
+
+/// @title PodAdminGateway for TRIBE Governance pods
+/// @notice Acts as a gateway for admin functionality and vetos in the TRIBE governance pod system
+/// @dev Contract is intended to be set as the podAdmin for all deployed Orca pods. It then allows
+///
+contract PodAdminGateway is CoreRef, IPodAdminGateway {
+    using EnumerableSet for EnumerableSet.Bytes32Set;
+
+    /// @notice Orca membership token for the pods. Handles permissioning pod members
+    IMemberToken private immutable memberToken;
+
+    /// @notice Pod factory which creates optimistic pods and acts as a source of information
+    IPodFactory private immutable podFactory;
+
+    constructor(
+        address _core,
+        address _memberToken,
+        address _podFactory
+    ) CoreRef(_core) {
+        require(_memberToken != address(0x0), "ZERO_ADDRESS");
+        require(_podFactory != address(0x0), "ZERO_ADDRESS");
+
+        memberToken = IMemberToken(_memberToken);
+        podFactory = IPodFactory(_podFactory);
+    }
+
+    ////////////////////////   GETTERS   ////////////////////////////////
+
+    /// @notice Calculate the specific pod admin role related to adding pod members
+    function getPodAddMemberRole(uint256 _podId) public pure returns (bytes32) {
+        return keccak256(abi.encode(_podId, "ORCA_POD", "POD_ADD_MEMBER_ROLE"));
+    }
+
+    /// @notice Calculate the pod admin role related to removing pod members
+    function getPodRemoveMemberRole(uint256 _podId)
+        public
+        pure
+        returns (bytes32)
+    {
+        return
+            keccak256(abi.encode(_podId, "ORCA_POD", "POD_REMOVE_MEMBER_ROLE"));
+    }
+
+    /// @notice Calculate the specific pod veto role, which allows an
+    function getPodVetoRole(uint256 _podId) public pure returns (bytes32) {
+        return keccak256(abi.encode(_podId, "ORCA_POD", "POD_VETO_ROLE"));
+    }
+
+    /// @notice Calculate the specific pod transfer admin role
+    function getPodTransferAdminRole(uint256 _podId)
+        public
+        pure
+        returns (bytes32)
+    {
+        return
+            keccak256(
+                abi.encode(_podId, "ORCA_POD", "POD_TRANSFER_ADMIN_ROLE")
+            );
+    }
+
+    /////////////////////////    ADMIN PRIVILEDGES       ////////////////////////////
+
+    /// @notice Transfer the pod admin address for a pod to another address
+    function transferPodAdmin(uint256 _podId, address newPodAdmin)
```

Comment:
It is. I exposed this functionality so that we can migrate to a different gateway contract easily in the future. It is available to a standard Orca pod admin, so feel it should be available via this gateway too